src/corehost/cli/ijwhost/exports.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
```

Comment:
I know that @AaronRobinsonMSFT  started this in the comhost, but I think I don't like the generic `exports.cpp` name of this file. It makes it hard to recognzie what I'm looking at in the editor, if nothing else. (It also means we have at least two files with the exact same name in the repo close to each other).

I would prefer to use the approach which `hostfxr` and `hostpolicy` already uses, that is `hostfxr.cpp` and `hostpolicy.cpp`. So in this case `ijwhost.cpp`. Can you please also rename the one in `comhost` to `comhost.cpp`?

Reply:
@vitek-karas As a historical note, when creating a C++ component I normally create a file called `exports.cpp` for a library and `main.cpp` for a console. These files represent contain the library/console contract for consumption. In the case of the console it will contain `main`/`wmain` per the OS contract and in the case of the library it will contain the exported functions for consumption. It makes it very clear to consumers of the source where to started when tracing code since the `exports.cpp` compilation unit will contain the only "public" functions.

Reply:
I understand that point of view... it's just that for me personally - I don't like having multiple files with the same names in one solution. But I can learn to live with that. This was more of a nit... so if I'm the only one... feel free to ignore.

Reply:
We should follow consistent style for all hosts in the repo; or ideally across .NET Core C++ components.

FWIW, the `exports.cpp` convention does not scale. It works only if the library exports small number of methods.

Reply:
@jkotas I disagree with it not scaling. Like many things "does not scale" is predicated on a specific design orthodoxy. I prefer libraries have small concise number of exports rather than a large number which makes unit testing and mental modelling simpler. In fact, if I recall the .NET Core system is/was trying to do just this by making many smaller libraries rather than large ones. We can argue over what large or small means, but it can work in practice - especially if you are creating a third party library. Creating complete generic platforms tends to fail due to wanting a single library, but like the `coreclr` or `hostfxr` libraries the number of exports should be kept small and concise because they are _not_ a generic platform API but simple library contracts that are intended to be configured and then activated.

@vitek-karas I am not pushing back, just offering why I generally do that for this kind of project. I absolutely understand your perspective and if that is your preference let's do it.

Reply:
I agree with you that creating small unit-testable components is a good thing. I disagree that there should be one shipping .dll per the small unit-testable component. We have tried to do that with .NET Core and it did not work well for us. It is hard to manage dependencies between the components when they are small .dlls and it costs performance to load many small .dlls.

> what large or small means

For me, small .dll is less than 1MB.